### PR TITLE
Add concept cluster creation functionality

### PR DESCRIPTION
**Pull request by fmatthies**
This pull requests adds the functionality to create concept clusters (basically graph structures that relate documents to concepts to phrases) in `Neo4j` from concept graphs that were created with the `concept-graphs-api`

### CODE REVIEW

File: src/main/java/care/smith/top/backend/model/neo4j/ConceptNodeEntity.java [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
 
+  @Id
   @Property("conceptId")
   private final String conceptId;
 
   @Property("labels")
   private final List<String> labels;
 
-  @Id @GeneratedValue private Long id;
   // ToDo: this takes a long time to load; maybe it's not necessary bc I get the Phrases later with
```
Comment (ChristophB) [inline src/main/java/care/smith/top/backend/model/neo4j/ConceptNodeEntity.java:18]: Please finish this TODO

File: src/main/java/care/smith/top/backend/repository/neo4j/PhraseNodeRepository.java [hunk 2]
```diff
@@ -19,4 +20,7 @@ public interface PhraseNodeRepository
           // ToDo: implement check for whether all or only exemplars should be regarded
```
Comment (ChristophB) [inline src/main/java/care/smith/top/backend/repository/neo4j/PhraseNodeRepository.java:20]: Please finish this TODO

File: src/main/java/care/smith/top/backend/service/nlp/ConceptClusterService.java [hunk 3]
```diff
@@ -1,55 +1,227 @@
...
+                      .getGraphStatisticsForProcess(processName)
+                      .getConceptGraphs())
+              .filter(conceptGraph -> graphIds.contains(conceptGraph.getId()));
+    }
+    Thread t =
+        new Thread(
+            () ->
+                conceptGraphStream.forEach(
+                    conceptGraph -> {
+                      // ToDo: threading of single graph creation seems to be not working; throws
```
Comment (ChristophB) [inline src/main/java/care/smith/top/backend/service/nlp/ConceptClusterService.java:185]: Please finish this TODO

File: src/test/java/care/smith/top/backend/service/nlp/ConceptClusterServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package care.smith.top.backend.service.nlp;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+class ConceptClusterServiceTest extends AbstractNLPTest {
+
+  @Test
```
Comment (ChristophB) [inline src/test/java/care/smith/top/backend/service/nlp/ConceptClusterServiceTest.java:9]: No test implementations

**Review by ChristophB — CHANGES_REQUESTED** [review]

File: src/main/java/care/smith/top/backend/model/neo4j/ConceptNodeEntity.java (hunk 1 above)
Comment (fmatthies) [reply src/main/java/care/smith/top/backend/model/neo4j/ConceptNodeEntity.java:18]: removed this TODO without changing anything. I need this to create the relations when generating the neo4j graphs.

File: src/main/java/care/smith/top/backend/service/nlp/ConceptClusterService.java (hunk 3 above)
Comment (fmatthies) [reply src/main/java/care/smith/top/backend/service/nlp/ConceptClusterService.java:185]: made an issue out of it

**Review by ChristophB — APPROVED** [review]

**Pull request merged**